wrappers/services.go

```diff
@@ -288,3 +342,61 @@ WantedBy={{.ServicesTarget}}
 
 	return templateOut.Bytes()
 }
+
+func genSocketFile(appInfo *snap.AppInfo, socketName string) []byte {
+	socketTemplate := `[Unit]
+# Auto-generated, DO NO EDIT
+Description=Socket {{.SocketName}} for snap application {{.App.Snap.Name}}.{{.App.Name}}
+Requires={{.MountUnit}}
+Wants={{.PrerequisiteTarget}}
+After={{.MountUnit}} {{.PrerequisiteTarget}}
+X-Snappy=yes
+
+[Socket]
+Service={{.ServiceFileName}}
+ListenStream={{.SocketInfo.ListenStream}}
+{{if .SocketInfo.SocketMode}}SocketMode={{.SocketInfo.SocketMode | printf "%04o"}}{{end}}
+
+[Install]
+WantedBy={{.SocketsTarget}}
+`
+	var templateOut bytes.Buffer
+	t := template.Must(template.New("socket-wrapper").Parse(socketTemplate))
+
+	wrapperData := struct {
+		App                *snap.AppInfo
+		ServiceFileName    string
+		PrerequisiteTarget string
+		SocketsTarget      string
+		MountUnit          string
+		SocketName         string
+		SocketInfo         *snap.SocketInfo
+	}{
+		App:                appInfo,
+		ServiceFileName:    filepath.Base(appInfo.ServiceFile()),
+		SocketsTarget:      systemd.SocketsTarget,
+		PrerequisiteTarget: systemd.PrerequisiteTarget,
+		MountUnit:          filepath.Base(systemd.MountUnitPath(appInfo.Snap.MountDir())),
+		SocketName:         socketName,
+		SocketInfo:         appInfo.Sockets[socketName],
+	}
+
+	if err := t.Execute(&templateOut, wrapperData); err != nil {
+		// this can never happen, except we forget a variable
+		logger.Panicf("Unable to execute template: %v", err)
+	}
+
+	return templateOut.Bytes()
+}
+
+func generateSnapSocketFiles(app *snap.AppInfo) (*map[string][]byte, error) {
```

Comment:
There's pretty much never a reason to have a pointer to a map in Go.

That said, we can drop this whole function. The code will be simpler if we call genServiceSocketFile in the exact spot where we need it above.